### ISSUE TITLE
.emacs.d/bin/doom: line 7: [: : integer expression expected

### ISSUE THREAD

**Issue by cxb811201** [hlissner/doom-emacs#3833]
**What actually happened?**
execute `doom sync`

.emacs.d/bin/doom: line 7: [: : integer expression expected

**System information:**
<details><pre>
((system
  (type . darwin)
  (config . "x86_64-apple-darwin19.6.0")
  (shell . "/bin/zsh")
  (uname . "Darwin 19.6.0 Darwin Kernel Version 19.6.0: Thu Jun 18 20:49:00 PDT 2020; root:xnu-6153.141.1~1/RELEASE_X86_64 x86_64")
  (path "/usr/local/opt/coreutils/libexec/gnubin" "/usr/local/opt/gawk/libexec/gnubin" "/usr/local/opt/findutils/libexec/gnubin" "/usr/local/opt/gnu-sed/libexec/gnubin" "/usr/local/opt/gnu-tar/libexec/gnubin" "/usr/local/opt/gnu-indent/libexec/gnubin" "/usr/local/opt/openssl/bin" "~/.go/bin" "~/.nvm/versions/node/v14.3.0/bin" "~/.cargo/bin" "~/.jenv/shims" "~/.jenv/bin" "~/.emacs.d/bin" "/usr/local/bin" "/usr/bin" "/bin" "/usr/sbin" "/sbin" "/Library/TeX/texbin" "/Library/Apple/usr/bin" "~/.antigen/bundles/robbyrussell/oh-my-zsh/lib" "~/.antigen/bundles/robbyrussell/oh-my-zsh/plugins/git" "~/.antigen/bundles/robbyrussell/oh-my-zsh/plugins/gitfast" "~/.antigen/bundles/robbyrussell/oh-my-zsh/plugins/colored-man-pages" "~/.antigen/bundles/robbyrussell/oh-my-zsh/plugins/history" "~/.antigen/bundles/robbyrussell/oh-my-zsh/plugins/z" "~/.antigen/bundles/robbyrussell/oh-my-zsh/plugins/extract" "~/.antigen/bundles/robbyrussell/oh-my-zsh/plugins/encode64" "~/.antigen/bundles/robbyrussell/oh-my-zsh/plugins/web-search" "~/.antigen/bundles/robbyrussell/oh-my-zsh/plugins/urltools" "~/.antigen/bundles/robbyrussell/oh-my-zsh/plugins/tmux" "~/.antigen/bundles/robbyrussell/oh-my-zsh/plugins/fzf" "~/.antigen/bundles/robbyrussell/oh-my-zsh/plugins/docker" "~/.antigen/bundles/robbyrussell/oh-my-zsh/plugins/docker-compose" "~/.antigen/bundles/robbyrussell/oh-my-zsh/plugins/brew" "~/.antigen/bundles/robbyrussell/oh-my-zsh/plugins/osx" "~/.antigen/bundles/robbyrussell/oh-my-zsh/plugins/cargo" "~/.antigen/bundles/djui/alias-tips" "~/.antigen/bundles/robbyrussell/oh-my-zsh/plugins/mvn" "~/.antigen/bundles/robbyrussell/oh-my-zsh/plugins/sbt" "~/.antigen/bundles/zdharma/fast-syntax-highlighting" "~/.antigen/bundles/zsh-users/zsh-autosuggestions" "~/.antigen/bundles/hlissner/zsh-autopair" "~/.antigen/bundles/romkatv/powerlevel10k" "/usr/local/Cellar/emacs-plus@28/28.0.50/libexec/emacs/28.0.50/x86_64-apple-darwin19.6.0"))
 (emacs
  (dir . "~/.emacs.d/")
  (version . "28.0.50")
  (build . "Aug 24, 2020")
  (buildopts . "--disable-dependency-tracking --disable-silent-rules --enable-locallisppath=/usr/local/share/emacs/site-lisp --infodir=/usr/local/Cellar/emacs-plus@28/28.0.50/share/info/emacs --prefix=/usr/local/Cellar/emacs-plus@28/28.0.50 --with-xml2 --with-gnutls --without-dbus --without-imagemagick --with-json --with-modules --with-rsvg --with-xwidgets --with-ns --disable-ns-self-contained")
  (features . "JPEG TIFF PNG RSVG GLIB NOTIFY KQUEUE ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS NS MODULES THREADS XWIDGETS JSON PDUMPER LCMS2")
  (traits batch server-running))
 (doom
  (dir . "~/.doom.d/")
  (version . "2.0.9")
  (build . "HEAD -> develop, origin/develop ce2cabf07 2020-08-25 06:07:47 -0400")
  (elc-files . 0)
  (modules :completion (company +childframe) (ivy +prescient +fuzzy +childframe +icons) :ui doom doom-quit hl-todo modeline nav-flash ophints (popup +all +defaults) treemacs vc-gutter window-select workspaces :editor file-templates fold format multiple-cursors snippets word-wrap :emacs dired electric (ibuffer +icons) undo vc :term vterm :checkers (syntax +childframe) (spell +aspell) :tools (debugger +lsp) docker editorconfig (eval +overlay) lookup (lsp +peek) magit make pdf :os macos :lang data emacs-lisp (go +lsp) json (java +lsp) (javascript +lsp) (markdown +grip) (org +dragndrop +hugo +pandoc +present +pomodoro +journal +pretty) plantuml rest (scala +lsp) sh (web +lsp) yaml :config (default +bindings +smartparens) :my (&user chinese +childframe) (&user ui) (&user editor) (&user calendar) (&user dired) (&user magit) (&user translate +childframe) (&user tools) (&user debugger) (&user lsp) (&user markdown) (&user plantuml) (&user protobuf) (&user go) (&user java +google-java-format) (&user scala) (&user sql) (&user keybindings))
  (packages "n/a")
  (unpin
   (:ui doom modeline)
   (:tools debugger lsp magit)
   (:lang java))
  (elpa "n/a")))

</pre></details>

**Comment by tecosaur** [comment]
If you could grab a backtrace with `doom -d sync` that could be helpful.

**Comment by tecosaur** [comment]
This looks like a duplicate of #3831.

**Comment by pszynk** [comment]
This error was introduced by the 63a03848a3e026498c2e25348b2288ccca2f7ad8 commit.

Lines:
```shell
:; $EMACS --no-site-file --script "$0" -- "$@" || __DOOMCODE=$?             # replaced ; with ||
:; [ "$__DOOMCODE" -eq 128 ] && { "$__DOOMPOST" "$0" "$@"; __DOOMCODE=$?; }
```

`__DOOMCODE` stays undefined if doom script returned 0. Not sure why `;` was switched with `||`.
Revert it, or use `${__DOOMCODE:-0}`?

**Comment by hlissner** [comment]
Duplicate of #3831. Fixed in 239516a.

**Comment by hlissner** [comment]
> __DOOMCODE stays undefined if doom script returned 0. Not sure why ; was switched with ||.

It was switched because `/bin/sh` implies `set -e` on most posix systems. So if `$EMACS --no-site-file --script "$0" -- "$@"` returns a non-zero exit code, the script is aborted, causing issues like #3746.
